js/pawn.js

```diff
@@ -0,0 +1,52 @@
+import { Piece } from "./piece.js"
+
+export class Pawn extends Piece{
+
+    constructor(pieceId, rowIndex, colIndex, playerId) {
+        super()
+
+        this.id = pieceId;
+        this.rowPos = rowIndex
+        this.colPos = colIndex;
+        this.prevRowPos = 0;
+        this.prevColPos = 0;
```

Comment:
You might want to remove the initialization of `prevRowPos` and `prevColPos` since they are initialized in the `Piece` constructor.